lib/spack/spack/environment/environment.py

```diff
@@ -1962,6 +2006,16 @@ def _concretize_from_constraints(spec_constraints, tests=False):
             invalid_constraints.extend(inv_variant_constraints)
 
 
+def _concretize_task(packed_arguments):
+    spec_constraints, tests = packed_arguments
+    try:
+        with tty.SuppressOutput(msg_enabled=False):
+            value = _concretize_from_constraints(spec_constraints, tests)
+    except Exception:
+        return spack.util.parallel.ErrorFromWorker()
+    return value
```

Comment:
Just a legibility idea, is this method easier to read this way?

```suggestion
    try:
        with tty.SuppressOutput(msg_enabled=False):
            return _concretize_from_constraints(spec_constraints, tests)
    except Exception:
        return spack.util.parallel.ErrorFromWorker()
```

Reply:
I'll try to have a single return value at the end.

Reply:
Done in 20cfb9c